### PR TITLE
Fix sharing scalar bug

### PR DESCRIPTION
**Pull request by pengwa**
If an initializer is used as graph outputs, we should keep its name, instead of renaming it as constant sharing transformer did currently.

To fix https://github.com/microsoft/onnxruntime/issues/14488

### CODE REVIEW

**Comment by RyanUnderhill** [comment]
I tested the fix and verified it fixes the linked issue.

File: onnxruntime/test/testdata/transform/scalar_const_not_share.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import numpy as np
+import onnx
+import onnxscript
+from onnx import numpy_helper
+from onnxscript import opset17 as op
+
+import onnxruntime
```
[the remark this answers is not in the document]
Comment (pengwa) [reply onnxruntime/test/testdata/transform/scalar_const_not_share.py]: Yeah, I fixed it in the latest commit.

**Review by RyanUnderhill — DISMISSED** [review]
What I understand looks good, hopefully a graph expert can double check.

**Review by skottmckay — APPROVED** [review]

**Review by askhade — APPROVED** [review]
LGTM

**Pull request merged**